components/dashboard/ActionPlan/ActionPlanBar.js

```diff
@@ -0,0 +1,218 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import Box from '@material-ui/core/Box';
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import NextLink from 'next/link';
+import Paper from '@material-ui/core/Paper';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Typography from '@material-ui/core/Typography';
+
+import AssignmentTurnedInIcon from '@material-ui/icons/AssignmentTurnedIn';
+import NextWeekIcon from '@material-ui/icons/NextWeek';
+import VpnKeyIcon from '@material-ui/icons/VpnKey';
+import FirebasePropTypes from '../../Firebase/PropTypes';
+import { useAuth } from '../../Auth';
+
+const PLAN_COLORS = {
+  goal: '#1980c8',
+  activity: '#f29a38',
+  application: '#7ea94f',
+};
+
+const ACTION_PLAN_DEFAULT = {
+  goals: 3,
+  activities: 3,
+  applications: 3,
+};
+
+const useStyles = makeStyles(theme => ({
+  paper: {
+    marginTop: theme.spacing(2),
+    padding: theme.spacing(2, 2, 2),
+    [theme.breakpoints.up('sm')]: {
+      padding: theme.spacing(2, 2, 2),
+    },
+    marginBottom: theme.spacing(8),
+  },
+  iconContainer: {
+    border: `3px solid`,
+    borderRadius: '50%',
+    lineHeight: 0,
+    padding: theme.spacing(1),
+  },
+  buttonLabel: {
+    justifyContent: 'left',
+    margin: theme.spacing(0.5, 3, 0.5),
+  },
+  text: {
+    textTransform: 'none',
+    textAlign: 'left',
+    marginLeft: theme.spacing(1),
+  },
+}));
+
+function ActionPlanBar({ userStats, actionPlan }) {
+  const classes = useStyles();
+  const { userDocRef } = useAuth();
+
+  function startOfWeek() {
```

Comment:
The `date-fns` library actually has a `startOfWeek()` function that might be worth looking at! [HERE](https://date-fns.org/v2.13.0/docs/startOfWeek)

Reply:
nice, use startOfWeek() from date-fns instead.